src/main/java/drukier/maze/Maze.java

```diff
@@ -0,0 +1,170 @@
+package drukier.maze;
+
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Random;
+import java.util.Stack;
+
+public class Maze {
+
+    private int mazeHeight = 10;
+    private int mazeWidth = 10;
+
+    MazeCell start;
+
+
+    MazeCell[][] array;
+
+    int visitedCells = 0;
+    int totalCells = getMazeHeight() * getMazeWidth();
+
+
+//    public Maze(int length, int width) {
+//        mazeHeight = length;
+//        mazeWidth = width;
+//
+//        maze = new MazeCell[mazeHeight][mazeWidth];
+//
+//    }
+
+    public int getMazeHeight() {
+        return mazeHeight;
+    }
+
+    public void setMazeHeight(int mazeHeight) {
+        this.mazeHeight = mazeHeight;
+    }
+
+    public int getMazeWidth() {
+        return mazeWidth;
+    }
+
+    public void setMazeWidth(int mazeWidth) {
+        this.mazeWidth = mazeWidth;
+    }
+
+    public MazeCell getStart() {
+        return start;
+    }
+
+    public void setStart(MazeCell start) {
+        this.start = start;
+    }
+
+    public static void main(String[] args) {
+
+        Maze maze = new Maze();
+        maze.mazeGenerator();
+    }
+
+    public MazeCell[][] mazeGenerator() {
+
+        Stack<MazeCell> visit = new Stack<MazeCell>();
+
+        start = findStartCell();
+
+        MazeCell current = start;
+
+        while (visitedCells < totalCells) {
+
+            //get neighbors
+            current = getNextCell(current);
+
+            // Chooses a neighbor to visit
+            // current = chooseNext();
+
+            if (!current.isVisited()) {
+                visit.push(current);
+                visitedCells++;
+                current.setVisited(true);
+                getNextCell(current);
+            }
+
+            if (current.isVisited()) {
+                getNextCell(current);
+            }
+
+        }
+        return array;
+
+    }
+
+    private MazeCell findStartCell() {
+        Random currentX = new Random();
```

Comment:
create one Random object as a constant, and use it everywhere you need a random number